Sources/AsyncQueue/ActorQueue.swift

```diff
@@ -80,65 +78,78 @@ public final class ActorQueue {
-    public func async(_ task: @escaping @Sendable () async -> Void) {
-        taskStreamContinuation.yield(task)
+    public func async(_ task: @escaping @Sendable (isolated ActorType) async -> Void) {
+        taskStreamContinuation.yield(ActorTask(target: target!, task: task))
     }
 
     /// Schedules an asynchronous task and returns after the task is complete.
     /// The scheduled task will not execute until all prior tasks have completed or suspended.
     /// - Parameter task: The task to enqueue.
     /// - Returns: The value returned from the enqueued task.
-    public func await<T>(_ task: @escaping @Sendable () async -> T) async -> T {
-        await withUnsafeContinuation { continuation in
-            taskStreamContinuation.yield {
-                continuation.resume(returning: await task())
-            }
+    public func await<T>(_ task: @escaping @Sendable (isolated ActorType) async -> T) async -> T {
+        let target = self.target! // Capture/retain the target before suspending.
+        return await withUnsafeContinuation { continuation in
+            taskStreamContinuation.yield(ActorTask(target: target) { target in
+                continuation.resume(returning: await task(target))
+            })
         }
     }
 
     /// Schedules an asynchronous throwing task and returns after the task is complete.
     /// The scheduled task will not execute until all prior tasks have completed or suspended.
     /// - Parameter task: The task to enqueue.
     /// - Returns: The value returned from the enqueued task.
-    public func await<T>(_ task: @escaping @Sendable () async throws -> T) async throws -> T {
-        try await withUnsafeThrowingContinuation { continuation in
-            taskStreamContinuation.yield {
+    public func await<T>(_ task: @escaping @Sendable (isolated ActorType) async throws -> T) async throws -> T {
+        let target = self.target! // Capture/retain the target before suspending.
+        return try await withUnsafeThrowingContinuation { continuation in
+            taskStreamContinuation.yield(ActorTask(target: target) { target in
                 do {
-                    continuation.resume(returning: try await task())
+                    continuation.resume(returning: try await task(target))
                 } catch {
                     continuation.resume(throwing: error)
                 }
-            }
+            })
         }
     }
 
     // MARK: Private
 
-    private let taskStreamContinuation: AsyncStream<@Sendable () async -> Void>.Continuation
+    private let taskStreamContinuation: AsyncStream<ActorTask>.Continuation
 
-    // MARK: - ActorExecutor
+    /// The target actor on whose isolated context our tasks run.
+    /// It is safe to use `unowned` here because it is API misuse to interact with an `ActorQueue` from an instance other than the `target`.
+    private unowned var target: ActorType?
 
-    private actor ActorExecutor {
```

Comment:
Now that we're explicitly targeting a specific actor, we no longer need another `actor` type! The code previously on `ActorExecutor` moved down into an extension on `Actor` below.